### PR TITLE
If using PowerShell, set exit code to that of command

### LINKED ISSUE #50: When using PowerShell on Windows, exit code of process is not set correctly
Using PowerShell, the exit code is always 0, regardless of the exit code of the command that was run: https://github.com/PrefectHQ/prefect-shell/blob/63e59fa10639c3472edbf71c6fca9a46fdb72042/prefect_shell/commands.py#L94

For example, this breaks the functionality of using `dbt test` with `trigger_dbt_cli_command` in `prefect-dbt`.

### PR DESCRIPTION
**Pull request by costructuralist**
Closes #50 

- [x] This pull request includes tests or only affects documentation.
- [x] Summarized PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-shell/blob/main/CHANGELOG.md)

### CODE REVIEW

**Comment by ahuang11** [comment]
Hi @costructuralist. Thank you adding this feature! I am unfamiliar with Windows exit code; can you provide an example output of it? 

I am unsure if we should append it to all Windows commands since some users may be keying off the last line of the output to schedule their next task. Perhaps we could add a flag?

**Comment by costructuralist** [comment]
Hi @ahuang11 — for what it's worth, in my testing, the `lines` variable

https://github.com/PrefectHQ/prefect-shell/blob/63e59fa10639c3472edbf71c6fca9a46fdb72042/prefect_shell/commands.py#L85

contains the same data in both cases: with the PR modification and without, so that the return value of the function is unaffected.

Here is a concrete reference for Windows exit codes, but the specifics don't really apply here: https://learn.microsoft.com/en-gb/windows/win32/debug/system-error-codes?redirectedfrom=MSDN

I'll post some example outputs from my testing shortly.

**Comment by costructuralist** [comment]
In both cases, `dbt test` is failing and returning an exit code of 1.

Without PR modification:

```
Shell command: ['powershell', 'C:\\Users\\user\\AppData\\Local\\Temp\\prefect-yvlarbtb.ps1']
Exit code: 0
Lines: ['08:06:39  Running with dbt=1.2.2', '08:06:39  Found 85 models, 17 tests, 1 snapshot, 0 analyses, 490 macros, 2 operations, 22 seed files, 37 sources, 0 exposures, 0 metrics', <snip>
'08:06:42    Got 1 result, configured to fail if != 0', '08:06:42', '08:06:42    compiled SQL at target\\compiled\\debut\\models\\sources.yml\\source_accepted_values_staging_8d7861efbdb88e8e7c2a955c0c94726f.sql', '08:06:42', '08:06:42  Done. PASS=15 WARN=0 ERROR=2 SKIP=0 TOTAL=17']
```

With PR modification:

```
Shell command: ['powershell', 'C:\\Users\\user\\AppData\\Local\\Temp\\prefect-ng10b20w.ps1']
Exit code: 1
Lines: ['08:08:36  Running with dbt=1.2.2', '08:08:36  Found 85 models, 17 tests, 1 snapshot, 0 analyses, 490 macros, 2 operations, 22 seed files, 37 sources, 0 exposures, 0 metrics', <snip>
'08:08:40    Got 1 result, configured to fail if != 0', '08:08:40', '08:08:40    compiled SQL at target\\compiled\\debut\\models\\sources.yml\\source_accepted_values_staging_8d7861efbdb88e8e7c2a955c0c94726f.sql', '08:08:40', '08:08:40  Done. PASS=15 WARN=0 ERROR=2 SKIP=0 TOTAL=17']

Traceback (most recent call last):
  File "C:\Source\data\prefect\test.py", line 119, in <module>
    loop.run_until_complete(task)
  File "C:\Users\user\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 646, in run_until_complete
    return future.result()
  File "C:\Source\data\prefect\test.py", line 104, in shell_run_command
    raise RuntimeError(msg)
RuntimeError: Command failed with exit code 1
```

**Comment by ahuang11** [comment]
Thank you! I wonder if we can add a test here: https://github.com/PrefectHQ/prefect-shell/blob/main/tests/test_commands_windows.py?

Is there a no-dependency command to test this? Maybe could add dbt cli into requirements-dev.txt if not.

**Comment by madkinsz** [comment]
> Is there a no-dependency command to test this? Maybe could add dbt cli into requirements-dev.txt if not.

You should be able to test with `false` / `true` / `exit CODE` equivalents.

File: tests/test_commands_windows.py [hunk 1]
```diff
@@ -162,6 +162,21 @@ def test_flow():
...
+        return shell_run_command(
+            # ping ??? returns exit code 1
+            command="echo 'two'; echo 'lines'; ping ???", 
+            return_all=True,
+            shell="powershell")
+
+    with pytest.raises(RuntimeError):
+        result = test_flow()
+        assert result[0].rstrip() == "two"
+        assert result[1].rstrip() == "lines"
```
Comment (madkinsz) [inline tests/test_commands_windows.py:177]: When is the exception thrown? At `test_flow()`? These assertions will not run in that case which is misleading.

Comment (costructuralist) [reply tests/test_commands_windows.py]: It is misleading indeed.  This test should only check if a non-zero exit code raises an exception.  The integrity of the output is taken care of by other tests.  Fixed.

File: tests/test_commands_windows.py [hunk 2]
```diff
@@ -162,6 +162,17 @@ def test_flow():
...
 
 
+def test_shell_run_command_throw_exception_on_nonzero_exit_code():
+    @flow
+    def test_flow():
+        return shell_run_command(
+            command="ping ???", shell="powershell"  # ping ??? returns exit code 1
+        )
+
+    with pytest.raises(RuntimeError):
```
Comment (desertaxle) [inline tests/test_commands_windows.py:172]: Can you also use the `match` argument of `pytest.raises` to make sure that the exception message contains the expected error code?

**Review by desertaxle — CHANGES_REQUESTED** [review]

**Review by ahuang11 — APPROVED** [review]

**Review by desertaxle — APPROVED** [review]

**Pull request merged**